### PR TITLE
Fix sniffer

### PR DESCRIPTION
**Pull request by greg-1-anderson**
Follow-on to #289.

### CODE REVIEW

**Comment by greg-1-anderson** [comment]
This does not work perfectly; the exit code from phpcs is 0 when no errors or warnings are found, and 1 if there are fixable or non-fixable errors. The result of this is that we prompt the user and ask if they'd like to fix the sniffer errors whether or not any of them are fixable.

**Comment by greg-1-anderson** [comment]
I don't really have time to address this myself at the moment, but I opened https://github.com/squizlabs/PHP_CodeSniffer/issues/930 to track the status code issue.

File: RoboFile.php [hunk 1]
```diff
@@ -352,14 +352,13 @@ public function trySuccess()
      *
      * @param string $file
      *    A file or directory to analyze.
-     * @param bool $autofix
-     *    Whether to run the automatic fixer or not.
+     * @option $autofix Whether to run the automatic fixer or not.
      */
-    public function sniff($file = 'src/', $autofix = false)
+    public function sniff($file = 'src/', $options = ['autofix' => false])
```
Comment (dustinleblanc) [inline RoboFile.php]: :+1:

**Comment by greg-1-anderson** [comment]
I added a --strict option, and defaulted the 'sniff' command to show only errors.  Now, `robo sniff` does not report any problems when run against the Robo source code.

**Comment by greg-1-anderson** [comment]
It is really useful to have `./robo sniff` run cleanly, so I am going to merge this. We can continue to adjust in follow-on issues as needed.

**Pull request merged**